subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/VerifyArtifactsStep.kt

```diff
@@ -51,13 +51,12 @@ open class VerifyArtifactsStep(
             entries.set(project.files(artifactsConfig.outputs.values.map { it.path }))
 
             project.gradle.buildFinished { buildResult ->
-                val isCopyScheduled = buildResult.gradle?.isTaskScheduled(this) ?: false
-                // TODO: consider skipped state (--dry-run)
-                val copyFinished = this.didWork
-                if (isCopyScheduled && !copyFinished) {
-                    project.ciLogger.info("Artifacts copying was scheduled but didn't run. Start it now")
-                    if (!this.didWork) {
-                        this.doAction()
+                if (buildResult.failure != null && buildResult.action == "Build") {
+                    if (!didWork) {
```

Comment:
What if we failed on the copy task?

Reply:
Build is already failed but you'll see a message from copy task too